Lib/multiprocessing/resource_tracker.py

```diff
@@ -283,9 +309,28 @@ def main(fd):
     try:
         # keep track of registered/unregistered resources
         with open(fd, 'rb') as f:
-            for line in f:
+            for raw in f:
                 try:
-                    cmd, name, rtype = line.strip().decode('ascii').split(':')
+                    line = raw.rstrip(b'\n')
+                    try:
+                        obj = json.loads(line.decode('ascii'))
+                    except Exception as e:
+                        raise ValueError("malformed resource_tracker message: %r" % (line,)) from e
+
+                    cmd = obj.get("cmd")
+                    rtype = obj.get("rtype")
+                    b64  = obj.get("base64_name")
```

Comment:
You don't need `get` with constant arguments:
```suggestion
                    cmd = obj["cmd"]
                    rtype = obj["rtype"]
                    b64  = obj["base64_name"]
```

Reply:
Alternately, make some fields optional -- then the PROBE message can be shorter:
```suggestion
                    cmd = obj["cmd"]
                    rtype = obj["rtype"]
                    b64  = obj.get("base64_name", "")
```